packages/integration-tests/src/entities/BookReview.ts

```diff
@@ -1,7 +1,30 @@
-import { BookReviewCodegen, bookReviewConfig, BookReviewOpts } from "./entities";
-import { EntityManager } from "joist-orm";
+import { Author, BookReviewCodegen, bookReviewConfig, BookReviewOpts } from "./entities";
+import { CustomReference, EntityManager, Reference } from "joist-orm";
 
 export class BookReview extends BookReviewCodegen {
+  public setFromOpts = false;
+  readonly author: Reference<BookReview, Author, never> = new CustomReference<
+    BookReview,
+    Author,
+    { book: "author" },
+    never
+  >(this, "author", {
+    load: async (review) => {
+      await review.load((r) => r.book.author);
+    },
+    get: (review) => review.book.get.author.get,
+    set: (review, author) => {
+      review.book.get.author.set(author);
+    },
+    setFromOpts: (review, author) => {
```

Comment:
Can we skip `setFromOpts` being a hook and have CustomReference just implement that internally and call `.set`?

I think `setFromOpts` means special things for other collections, but maybe/probably/hopefully not for our current CustomReference usage?

Reply:
will remove it for now